tests/test_utils.py

```diff
@@ -486,3 +492,167 @@ def template(x):
             template(3)
 
         assert str(recorder) == expected_output
+
+
+@qml.template
+def dummy_template(wires):
+    """Dummy template for inv tests."""
+    for wire in wires:
+        qml.RX(1, wires=[wire])
+        qml.RY(-1, wires=[wire])
+
+
+def inverted_dummy_template_operations(wires):
+    """The expected inverted operations for the dummy template."""
+    return sum(
+        [
+            [qml.RY(-1, wires=[wire]).inv(), qml.RX(1, wires=[wire]).inv()]
```

Comment:
It might be worth adding several operations of the same type as well, perhaps with alternating order as well